pkg/collector/corechecks/servicediscovery/module/comm_test.go

```diff
@@ -109,24 +114,38 @@ func TestShouldIgnoreComm(t *testing.T) {
 
 	serverBin := buildTestBin(t)
 	serverDir := filepath.Dir(serverBin)
+	discovery := newDiscovery()
+	require.NotEmpty(t, discovery)
+	require.NotEmpty(t, discovery.config.ignoreComms)
+	require.Equal(t, len(discovery.config.ignoreComms), 10)
 
 	for _, test := range testCases {
 		t.Run(test.name, func(t *testing.T) {
 			ctx, cancel := context.WithCancel(context.Background())
 			t.Cleanup(func() { cancel() })
 
 			makeAlias(t, test.comm, serverBin)
+			t.Cleanup(func() {
+				os.Remove(test.comm)
+			})
 			bin := filepath.Join(serverDir, test.comm)
 			cmd := exec.CommandContext(ctx, bin)
 			err := cmd.Start()
 			require.NoError(t, err)
+			t.Cleanup(func() {
+				_ = cmd.Process.Kill()
+			})
+
+			var proc *process.Process
+			require.EventuallyWithT(t, func(collect *assert.CollectT) {
+				proc, err = customNewProcess(int32(cmd.Process.Pid))
+				assert.NoError(collect, err)
+			}, 2*time.Second, 100*time.Millisecond)
 
 			require.EventuallyWithT(t, func(collect *assert.CollectT) {
-				proc, err := customNewProcess(int32(cmd.Process.Pid))
-				require.NoError(t, err)
-				ignore := shouldIgnoreComm(proc)
-				require.Equal(t, test.ignore, ignore)
-			}, 30*time.Second, 100*time.Millisecond)
+				ignore := discovery.shouldIgnoreComm(proc)
+				assert.Equal(t, test.ignore, ignore)
+			}, 200*time.Millisecond, 100*time.Millisecond)
```

Comment:
(give it enough time to run multiple times)
```suggestion
			}, 500*time.Millisecond, 100*time.Millisecond)
```

Reply:
done